tests/trainer/test_signal_handling.py

```diff
@@ -0,0 +1,66 @@
+import os
+import platform
+import signal
+
+import pytest
+import torch.distributed
+
+from pytorch_lightning import Trainer, Callback
+from pytorch_lightning.trainer.states import TrainerState
+from tests.base import EvalModelTemplate
+
+
+class KillCallback(Callback):
+    """ A callback that simulates a terminal signal (e.g. SIGINT). """
+
+    def __init__(self, signal_code):
+        self._signal = signal_code
+
+    def on_batch_end(self, trainer, pl_module):
+        # send the signal after the first batch
+        assert trainer.global_step == 0, "did not interrupt training after first batch"
+        pid = os.getpid()
+        os.kill(pid, self._signal)
+
+    def on_train_end(self, trainer, pl_module):
+        assert trainer.train_loop._teardown_already_run
+
+    def on_keyboard_interrupt(self, trainer, pl_module):
+        assert trainer.state == TrainerState.INTERRUPTED
+        assert trainer.interrupted
+
+
+def get_available_signal_codes():
+    codes = [signal.SIGINT]
+    if platform.system() != "Windows":
+        codes += [signal.SIGTERM, signal.SIGSEGV]
+    codes = [pytest.param(c) for c in codes]
+    return codes
+
+
+@pytest.mark.skipif(not torch.distributed.is_available(), reason="test requires torch.distributed module")
```

Comment:
does this run also on multi-pu or only multi node?

Reply:
this test here is for ddp_cpu, simulates ddp. 
the test below that is for multi-gpu
but there is no test for multi-node, and I don't know if it works for multi-node.